components/dashboard/src/repositories/list/RepositoryList.tsx

```diff
@@ -4,53 +4,41 @@
  * See License.AGPL.txt in the project root for license information.
  */
 
-import { FC, useCallback, useEffect, useState } from "react";
+import { FC, useCallback, useState } from "react";
 import { LoaderIcon } from "lucide-react";
 import { useHistory } from "react-router-dom";
 import { RepositoryListItem } from "./RepoListItem";
 import { useListConfigurations } from "../../data/configurations/configuration-queries";
-import { useStateWithDebounce } from "../../hooks/use-state-with-debounce";
 import { TextInput } from "../../components/forms/TextInputField";
-import { TextMuted } from "@podkit/typography/TextMuted";
 import { PageHeading } from "@podkit/layout/PageHeading";
 import { Button } from "@podkit/buttons/Button";
 import { useDocumentTitle } from "../../hooks/use-document-title";
-import { PaginationControls, PaginationCountText } from "./PaginationControls";
 import { Table, TableBody, TableHead, TableHeader, TableRow } from "@podkit/tables/Table";
 import { ImportRepositoryModal } from "../create/ImportRepositoryModal";
 import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+import { useQueryParams } from "../../hooks/use-query-params";
 
 const RepositoryListPage: FC = () => {
     useDocumentTitle("Imported repositories");
 
     const history = useHistory();
 
-    // TODO: Move this state into url search params
-    const [currentPage, setCurrentPage] = useState(1);
-    const [searchTerm, setSearchTerm, debouncedSearchTerm] = useStateWithDebounce("");
-
-    // Reset to page 1 when debounced search term changes (when we perform a new search)
-    useEffect(() => {
-        setCurrentPage(1);
-    }, [debouncedSearchTerm]);
-
-    // Have this set to a low value for now to test pagination while we develop this
-    // TODO: move this into state and add control for changing it
-    const pageSize = 5;
+    // Search/Filter params tracked in url query params
+    const params = useQueryParams();
```

Comment:
I went with letting the URL hold the state here. It seems to work well, but I could see us wanting to tweak this some as we add more filters that we put in the URL as well. I think we could abstract a hook to make it easier once we add some filter/sort options here.